src/abstractdataframe/show.jl

```diff
@@ -115,13 +138,12 @@ function getmaxwidths(df::AbstractDataFrame,
                 maxwidth = max(maxwidth, undefstrwidth)
             end
         end
-        maxwidths[j] = maxwidth
+        maxwidths[j] = max(maxwidth, ourstrwidth(compacttype(eltype(col))))
```

Comment:
Actually we have to change `textwidth` to `ourstrwidth` everywhere because of such beasts as (in general C escapes can sneak into column names or types - probably not realistic but possible):

this would be wrong with `textwidth`:

```julia> DataFrame(Symbol("\n\n\n\n\n") => [1,2,3])
3×1 DataFrame
│ Row │ \n\n\n\n\n │
│     │ Int64      │
├─────┼────────────┤
│ 1   │ 1          │
│ 2   │ 2          │
│ 3   │ 3          │
```

and (this is one character to wide with `textwidth`)

```
julia> struct C{T}
       v::Int
       end

julia> Base.show(io::IO, c::C) = print(io, c.v)

julia> DataFrame(x=[C{'\n'}(1)])
1×1 DataFrame
│ Row │ x        │
│     │ C{'\n'}  │
├─────┼──────────┤
│ 1   │ 1        │
```